src/form/Form.d.ts

```diff
@@ -0,0 +1,118 @@
+/// <reference types="react" />
+import { FieldValues, setValues } from '../utils/init';
+
+export interface ElementProps {
+  [fieldId: string]: {
+    [propName: string]: string;
+  };
+}
+
+type SetStep = (stepKey: string) => void;
+// am I missing trigger actions?
+type TriggerAction =
+  | 'field'
+  | 'click'
+  | 'change'
+  | 'addressSelect'
+  | 'load'
+  | 'blur';
+type Trigger = {
+  action: TriggerAction;
+  id: string;
+  text: string;
+  type: string; // we should maybe make a type for the element types
+};
+type IntegrationData = { firebaseAuthId?: string }; // this typing is incomplete
+type FormattedValues = {
+  displayText: string;
+  type: string;
+  value: FieldValues;
+};
+
+interface Context {
+  setValues: typeof setValues;
+  setOptions: (newOptions: { [servarKey: string]: FieldValues }) => void;
+  setProgress: (val: number) => void;
+  setStep: (stepKey: string) => boolean;
+  step: { style: { backgroundColor: string } };
+  userId: string;
+  stepName: string;
+}
+
+interface ContextOnChange extends Context {
+  changeKeys: string[];
+  integrationData: IntegrationData;
+  trigger: TriggerAction;
+  fields: FormattedValues;
+  lastStep: boolean;
+  elementRepeatIndex: number;
+  valueRepeatIndex: number;
+}
+
+interface ContextOnLoad extends Context {
+  fields: FormattedValues;
+  stepName: string;
+  previousStepName: string;
+  lastStep: boolean;
+  setStep: SetStep;
+  firstStepLoaded: boolean;
+  integrationData: IntegrationData;
+}
+
+interface ContextOnSubmit extends Context {
+  // I haven't used the plaid integration so not quite sure how to type submitFields
+  // does including Plaid field values change the type or is it still fine as FormattedValues
+  submitFields: FormattedValues;
```

Comment:
not sure about this type

Reply:
yeah plaid fields are a slightly different structure since they're hidden fields. normal formatted field payloads are { displayText, value, type} while hidden field payload is just value. wonder if we should maybe structure it so hidden fields are more consistent with form fields? or shoudl we just update the typings